src/array/array.cc

```diff
@@ -0,0 +1,372 @@
+/*!
+ *  Copyright (c) 2019 by Contributors
+ * \file array/array.cc
+ * \brief DGL array utilities implementation
+ */
+#include <dgl/array.h>
+#include "../c_api_common.h"
+#include "./array_op.h"
+#include "./arith.h"
+#include "./common.h"
+
+namespace dgl {
+
+using runtime::NDArray;
+
+namespace aten {
+
+IdArray NewIdArray(int64_t length, DLContext ctx, uint8_t nbits) {
+  return IdArray::Empty({length}, DLDataType{kDLInt, nbits, 1}, ctx);
+}
+
+BoolArray NewBoolArray(int64_t length, DLContext ctx) {
+  return BoolArray::Empty({length}, DLDataType{kDLInt, 64, 1}, ctx);
+}
```

Comment:
why not using a small number of bits here?

Reply:
It's a workaround. First, pytorch does not have bool array. Second, the return will be catched by `utils.Index` which assumes the input is an int64 id array. Since this method is used only in `HasEdgesBetween` and thus in `CSRIsNonZero`, I changed the implementation:
(1) This function is removed.
(2) Instead, when creating the return array, it will has the same bit width as the input CSR matrix.